lib/plugin/videoroom.ex

```diff
@@ -31,38 +42,371 @@ defmodule Janus.Plugin.VideoRoom do
+      ) do
+    message =
+      %{request: "allowed", allowed: allowed_list, room: room_id, action: action}
+      |> put_if_not_nil(@secret_key, room_secret)
+      |> new_janus_message(handle_id)
+
+    with {:ok,
+          %{
+            "videoroom" => "success",
+            "room" => ^room_id,
+            "allowed" => resulting_allowed_list
+          }} <- Session.execute_request(session, message) do
+      {:ok, resulting_allowed_list}
+    else
+      {:ok, %{"error" => _message, "error_code" => @no_such_room_error, "videoroom" => "event"}} ->
+        {:error, Errors.error(@no_such_room_error)}
+
+      {:error, reason} ->
+        {:error, reason}
+    end
+  end
+
+  @doc """
+  Sends request to kick given user out of the room.
+
+  ## Arguments
+  * `session` - valid `Janus.Session` process to send request through
+  * `room_id` - id of targeted room
+  * `user_id` - id of user to kick
+  * `handle_id` - id of caller's handle
+  * `room_secret` - optional room secret when requested room is protected
+
+  ## Returns
+  * on success: `:ok`.
+  * on error: `{:error, reason}`.
+  """
+  @spec kick(Janus.Session.t(), room_id, user_id :: String.t(), handle_id, room_secret) ::
+          :ok | {:error, any}
+  def kick(session, room_id, user_id, handle_id, room_secret \\ nil) do
+    message =
+      %{request: "kick", id: user_id, room: room_id}
+      |> put_if_not_nil(@secret_key, room_secret)
+      |> new_janus_message(handle_id)
+
+    with {:ok, %{"videoroom" => "success"}} <- Session.execute_request(session, message) do
+      :ok
+    else
+      {:ok, %{"error" => _message, "error_code" => @no_such_feed_error, "videoroom" => "event"}} ->
+        # previously :no_such_user
+        {:error, Errors.error(@no_such_feed_error)}
+
+      {:ok, %{"error" => _message, "error_code" => @no_such_room_error, "videoroom" => "event"}} ->
+        {:error, Errors.error(@no_such_room_error)}
+
+      {:error, reason} ->
+        {:error, reason}
+    end
+  end
+
+  @doc """
+  Sends request to list participants of given room.
+
+  ## Arguments
+  * `session` - valid `Janus.Session` process to send request through
+  * `room_id` - id of targeted room
+  * `handle_id` - id of caller's handle
+
+  ## Returns
+  * on success: `{:ok, participants}`.
+  * on error: `{:error, reason}`.
+  """
+  @spec list_participants(Janus.Session.t(), room_id, handle_id) ::
+          {:ok, list(participant)} | {:error, any}
+  def list_participants(session, room_id, handle_id) do
+    message =
+      %{request: "listparticipants", room: room_id}
+      |> new_janus_message(handle_id)
+
+    with {:ok,
+          %{
+            "videoroom" => "participants",
+            "room" => ^room_id,
+            "participants" => participants
+          }} <-
+           Session.execute_request(session, message) do
+      {:ok, participants}
+    else
+      {:ok, %{"error" => _message, "error_code" => @no_such_room_error, "videoroom" => "event"}} ->
```

Comment:
I would rather prefer if there was one common error handler for all the calls since the errors are standardized. So I would imagine something like this
```
with {:ok, _} <- Session.execute_request(session, message) do
  //
end
|> handle_videoroom_error()
```